src/test_helpers/integration.cljs

```diff
@@ -59,4 +78,143 @@
+                   (promise-reject (ex-info "event happened in unexpected order"
+                                            {:event-ids   event-ids
+                                             :waiting-for @waiting-ids}
+                                            ::out-of-order-event-id)))))))))))))
+
+(defn setup-app
+  []
+  (legacy-test/init!)
+  (if (app-initialized)
+    (p/resolved ::app-initialized)
+    (do
+      (rf/dispatch [:app-started])
+      (wait-for [:profile/get-profiles-overview-success]))))
+
+(defn setup-account
+  []
+  (if (messenger-started)
+    (p/resolved ::messenger-started)
+    (do
+      (create-multiaccount!)
+      (-> (wait-for [:messenger-started])
+          (.then #(assert-messenger-started))))))
+
+(defn integration-test
+  "Runs `f` inside `cljs.test/async` macro in a restorable re-frame checkpoint.
+
+  Option `fail-fast?`, when truthy (defaults to true), will force the test
+  runner to terminate on any test failure. Setting it to false can be useful
+  during development when you want the rest of the test suite to run due to a
+  flaky test. Prefer to fail fast in the CI to save on time & resources.
+
+  When `fail-fast?` is falsey, re-frame's state is automatically restored after
+  a test failure, so that the next integration test can run from a pristine
+  state.
+
+  Option `timeout-ms` controls the total time allowed to run `f`. The value
+  should be high enough to account for some variability, otherwise the test may
+  fail more often.
+  "
+  ([test-name f]
+   (integration-test test-name
+                     {:fail-fast? true
+                      :timeout-ms default-integration-test-timeout-ms}
+                     f))
+  ([test-name {:keys [fail-fast? timeout-ms]} f]
+   (test/async
+     done
+     (let [restore-fn (rf/make-restore-fn)]
+       (log-headline test-name)
+       (-> (p/do (f done))
+           (p/timeout timeout-ms)
+           (p/catch (fn [error]
+                      (is (nil? error))
+                      (when fail-fast?
+                        (js/process.exit 1))))
+           (p/finally (fn []
+                        (restore-fn)
+                        (done))))))))
+
+;;;; Fixtures
+
+(defn fixture-logged
+  "Fixture to set up the application and a logged account before the test runs.
+  Log out after the test is done.
+
+  Usage:
+
+      (use-fixtures :each (h/fixture-logged))"
+  []
+  {:before (fn []
+             (test/async done
+               (p/do (setup-app)
+                     (setup-account)
+                     (done))))
+   :after  (fn []
+             (test/async done
+               (p/do (logout)
+                     (wait-for [::logout/logout-method])
+                     (done))))})
+
+(defn fixture-silence-reframe
+  "Fixture to disable most re-frame messages.
+
+  Avoid using this fixture for non-dev purposes because in the CI output it's
+  desirable to have more data to debug, not less.
```

Comment:
Do we have any way to tell whether the tests are run as part of CI? If yes, we could make sure they're never silenced during CI.

Reply:
The tests run as part of the CI. This is here https://github.com/status-im/status-mobile/blob/7171c7085dc04315c315652af27fe5053c56bbfe/ci/Jenkinsfile.tests#L65

Does that answer your question?

About silencing, in the future, if silencing too much is merged by mistake, we could certainly force all tests in the CI to use a minimum level of `debug`, among other tricks. A nice suggestion @clauxx